src/app/handler/pet/pet.handler.go

```diff
@@ -34,17 +38,27 @@ func NewHandler(service petSvc.Service, imageService imageSvc.Service, validate
 // @Failure 503 {object} dto.ResponseServiceDownErr "Service is down"
 // @Router /v1/pets/ [get]
 func (h *Handler) FindAll(c router.IContext) {
+	userID := c.UserID()
```

Comment:
If not logged in, c.UserID() will be empty, which will make ```h.likeService.FindByUserId(userID)``` throw error.
I think we need to check if c.UserID() is empty (even better, make ```isAuth()``` in ```context.go``` )
if yes, return with ```isLike``` of logged in user.
if no, you can make ```isLike``` false (unauthed users will always see pets as unliked).

Reply:
🫡